secret-store/secret_store.py

```diff
@@ -0,0 +1,145 @@
+from dataclasses import dataclass
+from sqlcipher3 import dbapi2 as sqlcipher
+import logging
+import os
+from typing import Optional, Union
+
+
+_logger = logging.getLogger(__name__)
+# logging.basicConfig(level=logging.INFO, format="%(message)s")
+
+
+@dataclass(frozen=True)
+class Table:
+    name: str
+    columns: list[str]
+
+
+SECRETS_TABLE = Table("secrets", ["user", "password"])
+CONFIG_ITEMS_TABLE = Table("config_items", ["item"])
+
+
+@dataclass(frozen=True)
+class Credentials:
+    user: str
+    password: str
+
+
+class Secrets:
+    def __init__(self, db_file: str, master_password: str) -> None:
+        self.db_file = db_file
+        self.master_password = master_password
+        self._con = None
+        self._cur = None
+
+    def close(self) -> None:
+        if self._cur is not None:
+            self._cur.close()
+            self._cur = None
+        if self._con is not None:
+            self._con.close()
+            self._con = None
+
+    def connection(self) -> sqlcipher.Connection:
+        if self._con is None:
+            if not os.path.exists(self.db_file):
+                _logger.info(f"Creating file {self.db_file}")
+            self._con = sqlcipher.connect(self.db_file)
+            self._use_master_password()
+            self.create_tables()
+        return self._con
+
+    def _use_master_password(self) -> None:
+        """
+        If database is unencrypted then this method encrypts it.
+        If database is already encrypted then this method enables to access the data.
+        """
+        if self.master_password is not None:
+            sanitized = self.master_password.replace("'", "\\'")
+            self.cursor().execute(f"PRAGMA key = '{sanitized}'")
+
+    def cursor(self) -> sqlcipher.Cursor:
+        if self._cur is None:
+            self._cur = self.connection().cursor()
+        return self._cur
+
+    def has_table(self, name: str) -> bool:
+        res = self.cursor().execute("SELECT * FROM sqlite_master where name = ?", [name])
+        return True if res and res.fetchone() else False
+
+    # key represents a system, service, or application
+    def create_table(self, table: Table) -> None:
+        if self.has_table(table.name):
+            return
+        _logger.info(f'Creating table "{table.name}".')
+        columns = " ,".join(table.columns)
+        self.cursor().execute(f"CREATE TABLE {table.name} (key, {columns})")
+        self.connection().commit()
+
+    def create_tables(self) -> None:
+        for table in (SECRETS_TABLE, CONFIG_ITEMS_TABLE):
+            self.create_table(table)
```

Comment:
Is this worth the effort? I would prefer a simple SQL script with `create table` etc. This would also be more flexible.

Also: What happens if a table exists but has a different schema? Better keep it simple:
* If DB file does not exist: run SQL statements for creating schema
* If DB file exists: assume schema is present and do nothing.